### DISCUSSION TITLE
Zod doesn't pass data to input relation field. (create, connectOrCreate, connect fields)

### DISCUSSION THREAD

**Discussion by John-Hejzlar** [omar-dulaimi/prisma-trpc-generator#28] [Q&A] [answered]
Hi guys, shout out to @omar-dulaimi for this great lib.

Currently i am developing admin panel and using prisma-trpc-generator to generate all endpoints. I am trying to create relations between entities. So what i am doing is sending `{ connect: { id: 2 } }` on the relation field

# For example:
```
const { mutate, isLoading, isError, error } = trpc.useMutation([
    "post.createOnePost",
  ]);

  const sendData = async () => {
    mutate({
      data: {
        title: "Some generic title",
        slug: "some-generic-title",
        description: "asdf",
        openGraphImage: { connect: { id: 4 } },
        status: "DRAFT",
        category: { connect: { id: 1 } },
        coverImage: { connect: { id: 3 } },
        featured: false,
        publishedAt: new Date("2022-08-26T00:00:00.000Z"),
        author: "Nobody",
      },
    });
  };
```

But when i send this request, relation fields (openGraphImage, coverImage, category) are empty in input.
```
{
   data: {
    title: 'Some generic title',
     slug: 'some-generic-title',
    description: 'asdf',
    openGraphImage: {},
     status: 'DRAFT',
    category: {},
    coverImage: {},
    featured: false,
    publishedAt: 2022-08-26T00:00:00.000Z,
    author: 'Nobody'
  }
 }
```

I did some digging and looked into zod schema that is generated and i think the problem is this schema:
```
const Schema: z.ZodType<Prisma.ImageCreateNestedOneWithoutOpenGraphImageInput> =
  z.union([
    z.object({
      create: z
        .union([
          z.lazy(() => ImageCreateWithoutOpenGraphImageInputObjectSchema),
          z.lazy(
            () => ImageUncheckedCreateWithoutOpenGraphImageInputObjectSchema,
          ),
        ])
        .optional(),
    }),
    z.object({
      connectOrCreate: z
        .lazy(() => ImageCreateOrConnectWithoutOpenGraphImageInputObjectSchema)
        .optional(),
    }),
    z.object({
      connect: z.lazy(() => ImageWhereUniqueInputObjectSchema).optional(),
    }),
  ]);
```

Because zod can't distinguish between these 3 objects, it won't match any of these and leave the final field empty. What is weird that it doesn't throw any errors that the union is invalid or something.

Does anyone know if this is a bug or that i am sending data wrong way?

Thanks in advance

**Comment by omar-dulaimi** [comment, marked as answer]
Hey @John-Hejzlar 

I apologize for forgetting to respond. Glad that you like this project.

Also, please note that this issue should be fixed since `0.4.2`. So just updating to the latest version would fix it for you.

If any schema related problem arises, feel free to open a new issue here: https://github.com/omar-dulaimi/prisma-zod-generator
Since that's where the schemas are get generated!

Don't forget to star this project :)

**Comment by John-Hejzlar** [comment]
Just tested it and seems to be working correctly. Thanks a lot
